src/app/project/project-permissions/permissions-table-rows/permissions-table-rows.component.ts

```diff
@@ -46,6 +47,33 @@ export class PermissionsTableRowsComponent implements OnInit, TableComponent {
     this.currentProject = this.storageService.state.currentProject.data;
     this.entries = this.data.data;
     this.paginationData = this.data.paginationData;
+
+    this.ngxSmartModalService.getModal('confirmation-modal').onClose.subscribe(() => {
+      const modalData = this.ngxSmartModalService.getModalData('confirmation-modal');
+      if (modalData.deleteConfirm && modalData.user._id) {
+        // If the admin selected "OK" and a valid user ID is found, remove user from project
+        this.userService.removeUser(modalData.user)
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe(
+          (returnedUsers) => {
+            this.entriesVault = returnedUsers;
+            this.entriesVault = this.removeDuplicateUsers(this.entriesVault);
+            this.paginateUsers(this.paginationData.currentPage);
+          },
+          error => {
+            console.error(error);
+            alert('Uh-oh, couldn\'t remove user.');
+            this.router.navigate(['/p', this.currentProject._id ]);
+          },
+          () => { // onCompleted
+            this.openSnackBar(`${modalData.user.displayName || 'User'} has been removed from the permissions list.`, 'close')
+          })
+      } else if (modalData.deleteConfirm && !modalData.user._id) {
+        // If the admin selected "OK" but a valid user ID isn't found, trigger a console error
+        console.error('Unable to delete user. The user entry is malformed.');
+      }
+    });
+
   }
```

Comment:
Remember to clean up your onClose subscription in ngOnDestroy to avoid memory leaks!
```suggestion
  }

  ngOnDestroy(): void {
    this.ngUnsubscribe.next(true);
    this.ngUnsubscribe.complete();
  }

```

Reply:
Yes it was missing before and it does become more important when adding additional stuff to the ngOnInit.